cpp/src/binaryop/jit/code/kernel.cpp

```diff
@@ -41,14 +44,25 @@ R"***(
         int step = blksz * gridsz;
 
         for (gdf_size_type i=start; i<size; i+=step) {
-            out_data[i] = TypeOpe::template operate<TypeOut, TypeLhs, TypeRhs>(lhs_data[i], *reinterpret_cast<TypeRhs*>(&rhs_data));
+            gdf_size_type valid_index = i >> 3;
```

Comment:
Don't hard code dependencies on the mask representation. We have utilities for getting and setting bits in the mask. Also, what about setting the output bit mask?

Reply:
For the output bit mask, `binary_valid_mask_and` and `scalar_col_valid_mask_and` in `binaryop.cpp` handle that correctly  for non-comparison operators via `cudaMemCopy`, but comparison operators will never need an output bitmask, as the output of the operation is always a valid boolean. We can just bypass those functions if the operator is a comparison.

 - Does introduce the validity bit to the operator signature introduce a tangible overhead? If so, we could seperate out the comparison ops, since those bits are unnecessary for the other ops.
 - I wasn't able to find the bitmask utils. Where are they?

Reply:
Looking at this https://github.com/rapidsai/cudf/issues/2366#issuecomment-514824589, I realize that not every language considers comparisons between nulls as valid, and your point about the validity output should probably be implemented for comparison operations. I believe that would provide enough information to satisfy most/all (happy to debate this).